sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs

```diff
@@ -76,25 +53,26 @@ public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder bui
         /// <item>SQL Client.</item>
         /// </list>
         /// </remarks>
-        public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder builder, Action<AzureMonitorOptions> configureAzureMonitor)
+        public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder builder, Action<AzureMonitorOptions>? configureAzureMonitor = null, Action<ResourceBuilder>? configureResource = null)
```

Comment:
configuring Resource is something upstream OpenTelemetry already provides, and AzMon distro should not add anything new here.

Reply:
you're right, but see the linked issue; the code in AzMon stomps any configured resource builder.

Reply:
If AzureMonitorDistro does not respect user resource, I fully agree it needs to be fixed! (also, AzMon does not support Resource fully - only the subset from it to populate RoleName and Instance, and rest are ignored).

Maybe the issue is right in this doc itself:
https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-configuration?tabs=aspnetcore#set-the-cloud-role-name-and-the-cloud-role-instance

It is only applied to tracing alone, not for metrics, and **not for logs**!
It should be:
```
var resourceAttributes = new Dictionary<string, object> {
    { "service.name", "my-service" },
    { "service.namespace", "my-namespace" },
    { "service.instance.id", "my-instance" }};
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddOpenTelemetry().UseAzureMonitor().ConfigureResource(r => r.AddAttributes(resourceAttributes)); //applied to all 3
```

Please see if this is indeed just a doc issue.

Reply:
yes and no. I should be able to run `ConfigureResource` whenever I want and it should be respected. In the case of AzMon, it's not if I run ConfigureResource() *before* `UseAzureMonitor`

Reply:
That should be a fully internal fix, w/o any public api changes.
Does it work for traces,metrics and broken *only* for logging? (I'd assume so, if at all it is broken!). 

Adding new public api overload for accepting `Resource` (a pure otel concept), to fix a bug is not the direction I can agree.